### PR TITLE
Add Validations and Unit Tests for User, Post, Comment, and Like Models

### PR DESCRIPTION
**Pull request by gaks1**
# Description:
This pull request adds the necessary model validations and unit specifications to meet the project requirements. The following changes were made:

1. User Model:
    - Added a validation to ensure that the name is not blank.
    - Added a validation for the PostsCounter attribute, ensuring it is an integer greater than or equal to zero.
    - Added unit specs to validate the presence and format of the name attribute, as well as the PostsCounter attribute.
2. Post Model:
    - Added a validation to ensure that the title is not blank.
    - Added a validation to limit the title length to a maximum of 250 characters.
    - Added validations for the CommentsCounter and LikesCounter attributes, ensuring they are integers greater than or equal to zero.
    - Added unit specs to validate the presence, length, and format of the title attribute, as well as the CommentsCounter and LikesCounter attributes.
    - Added a method to retrieve the five most recent comments for a given post, along with unit specs to verify its functionality.
3. Comment and Like Models:
    - Created unit specs to test the incrementation of the comments_counter and likes_counter attributes upon creation.
4. RSpec Configuration:
    - Added the 'rspec-rails' gem to the Gemfile and updated Gemfile.lock accordingly.
    - Created spec files and added unit specs for Comment, Like, Post, and User models.
    - Updated the 'rails_helper.rb' and 'spec_helper.rb' files to configure RSpec and load necessary dependencies.

### CODE REVIEW

File: spec/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  it 'increments comments_counter after creation' do
+    comment = Comment.new(post:, author: user, text: 'Hello')
+    expect { comment.save }.to change { post.reload.comments_counter }.by(1)
+  end
+
+  it 'updates the comments counter on the post when saved' do
+    new_comment = Comment.new(post:, author: user, text: 'Hello')
+    expect { new_comment.save }.to change { post.reload.comments_counter }.by(1)
+  end
```
Comment (ishimwezachee) [inline spec/comment_spec.rb]: - [ ]  You did great job to implement the unit tests, but Comment model tests are not passing, I am getting an errors, and I recommend you to work on them and fix them. 
image
- [ ]  You should also consider the method for comment model `update_post_comments_counter` and test it as well to have your app fully Tested

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  it 'returns the 5 most recent comments for a given post' do
+    comments = Array.new(7) { |i| Comment.create(post:, author: user, text: "Comment #{i}") }
+    expect(post.recent_comments).to match_array(comments.last(5).reverse)
+  end
+
+  describe '#update_post_counter' do
+    it 'increments the post_counter of the author by one after creating a new post' do
+      expect { post.save }.to change { user.post_counter }.by(1)
+    end
```
Comment (ishimwezachee) [inline spec/post_spec.rb]: - [ ]  You should fix the syntax error encountered when running Post model tests
image

- [ ]  Please checkout on its  corresponding methods and tests them as well for your app to fully tested properly

File: spec/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  it 'increments likes_counter after creation' do
+    like = Like.new(post:, author: user)
+    expect { like.save }.to change { post.reload.likes_counter }.by(1)
+  end
+
+  it 'updates the likes counter on the post when saved' do
+    new_like = Like.new(post:, author: user)
+    expect { new_like.save }.to change { post.reload.likes_counter }.by(1)
+  end
```
Comment (ishimwezachee) [inline spec/like_spec.rb]: - [ ]  Like model and its corresponding methods are not properly tested, please fix the syntax error encountered when run the tests and  use the methods in your tests as well. 
image

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi  @gaks1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image
## Highlights
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @gaks1 
congratulations-1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Added Tests For Custom Methods As Requested :heavy_check_mark: 
Added Validation Tests :heavy_check_mark: 
Passing Test Cases :heavy_check_mark: 
image
Good Commit Messages :heavy_check_mark: 

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**